x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/remoteaccess/RemoteAccessAuthenticationServiceIntegTests.java

```diff
@@ -0,0 +1,229 @@
+                new RoleDescriptorsIntersection(new RoleDescriptor("role", null, null, null, null, null, null, null))
+            ).writeToContext(threadContext);
+
+            final AtomicReference<Authentication> actual = new AtomicReference<>();
+            final CountDownLatch latch = new CountDownLatch(1);
+            service.authenticate(ClusterStateAction.NAME, new SearchRequest(), false, new LatchedActionListener<>(new ActionListener<>() {
+                @Override
+                public void onResponse(Authentication authentication) {
+                    actual.set(authentication);
+                }
+
+                @Override
+                public void onFailure(Exception e) {
+                    fail();
+                }
+            }, latch));
+            latch.await();
+            final Authentication actualAuthentication = actual.get();
+            assertNotNull(actualAuthentication);
+            assertThat(actualAuthentication.getEffectiveSubject().getUser(), is(SystemUser.INSTANCE));
+            @SuppressWarnings("unchecked")
+            List<RemoteAccessAuthentication.RoleDescriptorsBytes> rds = (List<
+                RemoteAccessAuthentication.RoleDescriptorsBytes>) actualAuthentication.getAuthenticatingSubject()
+                    .getMetadata()
+                    .get(AuthenticationField.REMOTE_ACCESS_ROLE_DESCRIPTORS_KEY);
+            assertThat(rds.size(), equalTo(1));
+            assertThat(rds.get(0).toRoleDescriptors(), equalTo(Set.of(RemoteAccessAuthenticationService.CROSS_CLUSTER_INTERNAL_ROLE)));
+        }
+    }
+
+    private String getEncodedRemoteAccessApiKey() {
+        final CreateApiKeyResponse response = new CreateApiKeyRequestBuilder(client().admin().cluster()).setName("remote_access_key").get();
+        return Base64.getEncoder().encodeToString((response.getId() + ":" + response.getKey()).getBytes(StandardCharsets.UTF_8));
+    }
+
+    private void authenticateAndAssertExpectedFailure(RemoteAccessAuthenticationService service, Consumer<Exception> assertions)
+        throws InterruptedException {
+        final AtomicReference<Exception> actual = new AtomicReference<>();
+        final AtomicReference<Authentication> actualAuthentication = new AtomicReference<>();
+        final CountDownLatch latch = new CountDownLatch(1);
+        service.authenticate(SearchAction.NAME, new SearchRequest(), false, new LatchedActionListener<>(new ActionListener<>() {
+            @Override
+            public void onResponse(Authentication authentication) {
+                actualAuthentication.set(authentication);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                actual.set(e);
+            }
+        }, latch));
+        latch.await();
+        assertNull(actualAuthentication.get());
+        final Exception actualException = actual.get();
+        assertNotNull(actual);
+        assertions.accept(actualException);
```

Comment:
Why do we need `LatchedActionListener`? Doesn't `PlainActionListener` work just fine?

```java
var future = new PlaintActionFuture< Authentication>();
var e = expectThrows(Exception.class, () -> service.authenticate(SearchAction.NAME, new SearchRequest(), false, future));
assertions.accept(e);
```